script/domVisitor.ts

```diff
@@ -0,0 +1,124 @@
+// tslint:disable:no-console
+import * as dom from '@openstax/types/lib.dom';
+import { dirname } from 'path';
+import puppeteer from 'puppeteer';
+
+const rootUrl = `http://localhost:${process.env.PORT || '8000'}`;
+const devTools = false;
+const onlyOneBook = process.argv[3]; // because it's being called via entry.js
+
+function browserFindMatches(): string[] {
+  // Note: This executes in the browser context
+  if (!document) {
+    throw new Error(`BUG: Should run in browser context`);
+  }
+  function nearestId(el: dom.Element): string {
+    const id = el.getAttribute('id');
+    if (id) {
+      return id;
+    } else if (el.parentElement) {
+      return nearestId(el.parentElement);
+    }
+    throw new Error('BUG: Could not find an ancestor with an id');
+  }
+
+  // Remove all the MJX_Assistive_MathML elements because they
+  // can be wider even though they are not visible
+  document.querySelectorAll('.MJX_Assistive_MathML').forEach((el) => el.remove());
+
+  const wideIds = new Set<string>();
+  const root = document.querySelector('#main-content > div');
+  if (!root) {
+    throw new Error(`BUG: Could not find content root`);
+  }
+  const rootRight = root.getBoundingClientRect().right;
+  root.querySelectorAll('*').forEach((c) => {
+    if (c.getBoundingClientRect().right > rootRight) {
+      wideIds.add(nearestId(c));
+    }
+  });
+  return Array.from(wideIds);
+}
+
+async function visitPages(page: puppeteer.Page, bookHref: string) {
+  const bookPages = await findBookPages(page, bookHref);
+  let index = 0;
+  for (const bookPageUrl of bookPages) {
```

Comment:
since you're using `index` anyway, you could change this to `for (const [index, bookPageUrl] of bookPages.entries()) {` or `for (let index = 0; index < bookPages.length; index++) {`

Reply:
I just got rid of `index` 😄